app/models/user.rb

```diff
@@ -95,7 +95,15 @@ def active_identities
   end
 
   def active_profile
-    @active_profile ||= profiles.verified.find(&:active?)
+    if defined?(@active_profile)
+      if @active_profile && !@active_profile.active
+        remove_instance_variable(:@active_profile)
+      else
+        return @active_profile
+      end
+    end
+
```

Comment:
Do we need to explicitly `remove_instance_variable`, or would it be enough to let all but the "still active" state fall through to the fresh assignment below? I guess it does get kinda weird dealing with possible `nil` values.

```suggestion
    return @active_profile if defined?(@active_profile) && @active_profile&.active != false
```